src/olympia/addons/models.py

```diff
@@ -329,13 +329,16 @@ def get_mad_queue(self, admin_reviewer=False):
             # to have a version since they got the needs_human_review_by_mad
             # flag, so returning incomplete ones is acceptable.
             .filter(
-                status__in=[
+                Q(status__in=[
                     amo.STATUS_APPROVED, amo.STATUS_NOMINATED, amo.STATUS_NULL
-                ],
-                versions__files__status__in=[
+                ]),
+                Q(versions__files__status__in=[
                     amo.STATUS_APPROVED, amo.STATUS_AWAITING_REVIEW,
-                ],
-                versions__reviewerflags__needs_human_review_by_mad=True
+                ]),
+                (Q(versions__reviewerflags__needs_human_review_by_mad=True,
+                    versions__channel=amo.RELEASE_CHANNEL_UNLISTED) |
```

Comment:
This will be filtering and returning the addon if _any_ version is unlisted and any there are _any_ needs_human_review_by_mad set on the versions.  I.e. it's not testing that the same version is both unlisted and flagged. (if I'm remembering my queryset logic correctly). 

This addon would be in the queryset:
version_a: unlisted, needs_human_review_by_mad=False
version_b: listed, needs_human_review_by_mad=True
version_c (current_version): listed, needs_human_review_by_mad=False

 Just checking this is what you intended.  Can you mix in an unlisted version into the tests?

Reply:
That doesn't sound like what we want. What would the correct filter look like?

Reply:
Doesn't `Q(A, B)` mean `A AND B`?

Reply:
it does.  But it's `<any version of that addon>__reviewerflags__needs_human_review_by_mad` `AND` `<any version of that addon>__channel=amo.RELEASE_CHANNEL_UNLISTED`

Reply:
I see. How do you fix that?

Reply:
does forcing it into a subquery work?
```
.filter(
    status__in=[
        amo.STATUS_APPROVED, amo.STATUS_NOMINATED, amo.STATUS_NULL
    ],
    versions__files__status__in=[
        amo.STATUS_APPROVED, amo.STATUS_AWAITING_REVIEW,
    ],
    versions__in=Version.unfiltered.filter(
        Q(reviewerflags__needs_human_review_by_mad=True, channel=amo.RELEASE_CHANNEL_UNLISTED) |
        Q(addon___current_version_id=id)
    )
```

Reply:
I added this scenario in 3b16c96f1a and the current code seems to work based on that.